src/agora/worker.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file worker.h
+ * @brief Declaration file for the main worker class
+ */
+
+#ifndef WORKER_H_
+#define WORKER_H_
+
+#include <vector>
+
+#include "agora_helper.h"
+#include "buffer.h"
+#include "concurrent_queue_wrapper.h"
+#include "concurrentqueue.h"
+#include "config.h"
+#include "csv_logger.h"
+#include "dodecode.h"
+#include "dodemul.h"
+#include "doencode.h"
+#include "dofft.h"
+#include "doifft.h"
+#include "doprecode.h"
+#include "dozf.h"
+#include "mat_logger.h"
+#include "phy_stats.h"
+#include "stats.h"
+#include "symbols.h"
+
+class Worker {
```

Comment:
Let's not call this worker.  It is too generic as we have a billion type of workers.    Not sure if this is the time but creating a worker interface definition might be good.   Let's discuss

Reply:
Done